### ISSUE TITLE
CoreML export. How to parse MLMultiArray

### ISSUE THREAD

**Issue by obohrer** [ultralytics/yolov5#6123]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, I'm relatively new to this domain and currently experimenting with running CoreML to detect objects on IOS.
I've managed to train a custom model using yolov5 and tested the weights using a webcam as source. It does work 🚀.
However, when exporting the model to coreml using `export.py` it has been a struggle.
The exported model has the input dimension 480x640, so far so good.
However, the outputs are the following `var_875`, `var_860`.
During detection CoreML returns 2 matrices :`[1 x 3 x 80 x 60 x 6]` and  `[1 x 3 x 40 x 30 x 6]`
It seems that 80 x 60 and 40 x 30 are the same ratio as the dimension (*8,*16 multiplier)
Is there documentation on how to parse such outputs?

So far I've found many functions to parse those MLMultiArrayValue but none of them are producing correct scores or bounding boxes.

Any pointers on what those dimensions means and how to parse it would be greatly appreciated and happy to contribute back with code snippets / improve the exporter (eg name the outputs correctly)

**Comment by obohrer** [comment]
This repository has been useful: https://github.com/dbsystel/yolov5-coreml-tools
Had to tweak the coremltools dependency and skip one of the outputNames in yolov5-coreml-tools.
So far it seems that the bounding boxes are correct, same as predictions.
Only issue I'm seeing on iOS now is the confidence always show 100% (same as with models created with Create ML when there is one class)

**Comment by glenn-jocher** [comment]
@obohrer good news 😃! Your original issue may now be fixed ✅ in PR #6195. This PR adds support for YOLOv5 CoreML inference.

```python
!python export.py --weights yolov5s.pt --include coreml  # CoreML export
!python detect.py --weights yolov5s.mlmodel  # CoreML inference (MacOS-only)
!python val.py --weights yolov5s.mlmodle  # CoreML validation (MacOS-only)

model = torch.hub.load('ultralytics/yolov5', 'custom', 'yolov5s.mlmodel')  # CoreML PyTorch Hub model
```

<img width="1360" alt="Screen Shot 2022-01-04 at 5 41 07 PM" src="https://user-images.githubusercontent.com/26833433/148147262-1527cccc-19ca-43c4-b50a-c9a81cfd0bb5.png">

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by tcollins590** [comment]
> This repository has been useful: https://github.com/dbsystel/yolov5-coreml-tools Had to tweak the coremltools dependency and skip one of the outputNames in yolov5-coreml-tools. So far it seems that the bounding boxes are correct, same as predictions. Only issue I'm seeing on iOS now is the confidence always show 100% (same as with models created with Create ML when there is one class)

Would you be able to share your updated dependency and which outputNames you skipped? Thank you
